storage/consistent_watchable_store.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"encoding/binary"
+	"log"
+)
+
+var (
+	entIndexKeyName = []byte("entry_index")
+)
+
+// EntIndexGetter is an interface that wraps the Get method.
+// Entry index is a value in numerical order that identifies the entry
+// in a consistent replica log.
+type EntIndexGetter interface {
```

Comment:
Do not use abbr (Entry->Entry) for a public interface naming. The scope of the interface is large. Using abbr makes this less clear. Also I still like ConsistentIndex more. /cc I know @jonboulle is good at naming